### DISCUSSION TITLE
using lws for react front-end webapp builds?

### DISCUSSION THREAD

**Discussion by JamesKoenig** [lwsjs/local-web-server#185] [Q&A] [answered]
Hiya!   I've been using `local-web-server` as a dev tool while contributing to a foss webapp, given that it allows me to avoid CORS issues.

While working on things for it, I started to wonder about making a fork that used react to clean up and organize the codebase, and saw that the readme for `ws` specifically lists react as something it could be used as part of the build process for:
i.e. README.md, line 22:
> Build any type of front-end web application (static, dynamic, Single Page App, Progessive Web App, React etc).

I'm confused as to where to learn more about using local-web-server for building & developing front-end web applications more generally.  As it stands, the README is the only location react is mentioned throughout the entire project.

I realize that the question of implementing a project build might be outside the scope of `ws`'s intended documentation; For seemingly comparable tools, like `webpack`, that kind of thing tends to be relegated to random developer's blog posts; but, I was hoping for some insight about how to get underway with using `ws` in building a react application. 

Any insight is appreciated.  React's articles on adding it to an existing project want people to use Vite, which would likely replace `local-web-server` in my environment if I were to do that, and community articles on the topic tend to refer to using babel.

**Comment by 75lb** [comment, marked as answer]
You can host any website using local-web-server, whether using plain HTML, CSS and JS or a highly-opinionated DSL like React.. Just keep in mind local-web-server is just a web server, *not* a build tool.. You need to build your React site yourself using your regular tools, then the final output (the compiled html, css, js etc resources) can be hosted using local-web-server.. If your react app is a SPA then you'll probably need to use [`ws --spa`](https://github.com/lwsjs/local-web-server/wiki/How-to-serve-a-Single-Page-Application-(SPA))..

**Reply by JamesKoenig** [reply]
thank you!! I wasn't sure if I was misreading the source and the readme since it said build but that clarifies everything I saw 😊
